src/App.js

```diff
@@ -1,18 +1,44 @@
 import React, { Component } from 'react';
 import logo from './logo.svg';
 import './App.css';
+import InputItem from './InputItem';
+import TodoList from './TodoList';
 
 class App extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {data: []}
+  }
+  
+  recieveItem = (item) => {
+    this.state.data.push(item);
+    this.setState({data: this.state.data});
+  }
+
+  deleteItem = (item) => {
+    let new_data = this.state.data.filter((obj) => {
+      return obj.id !== item.id;
+    })
+    this.setState({data: new_data});
+  }
+
+  toggleItem = (item) => {
+    let selectedItem = this.state.data.find((object => object.id === item.id))
+    selectedItem.isChecked = !selectedItem.isChecked;
+    this.setState({data: this.state.data});
+    
+  }
+
   render() {
     return (
       <div className="App">
         <header className="App-header">
           <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
+          <h1 className="App-title">Todo App</h1>
         </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
+        <InputItem itemCallback = {this.recieveItem}/>
```

Comment:
itemCallback isnt a very good name. We should name them according to their action. In this case InputItem is either changing or submitting so onChange or onSubmit. An even better way would be to have a InputTodo which has the action to "add"